### PR TITLE
Add LiveDatabaseTestCase

### PR DESCRIPTION
**Pull request by stephencelis**
This audits our tests for ones that actually need the live database and avoids the IO of creating/migrating/dropping for a bunch of test cases.

### CODE REVIEW

**Comment by stephencelis** [comment]
Dunno why but Ubuntu CI started failing out of the blue :/ Can't repro locally

File: Tests/PointFreeTests/AccountTests/UpdateProfileTests.swift [hunk 1]
```diff
@@ -96,6 +96,13 @@ class UpdateProfileTests: TestCase {
     assertSnapshot(matching: output, as: .conn)
     #endif
   }
+}
+
+class UpdateProfileTests: LiveDatabaseTestCase {
```
Comment (mbrandonw) [inline Tests/PointFreeTests/AccountTests/UpdateProfileTests.swift]: should this inherit from `TestCase` instead?

File: Tests/PointFreeTests/AuthTests.swift [hunk 2]
```diff
@@ -10,7 +10,7 @@ import Optics
 import SnapshotTesting
 import XCTest
 
-class AuthTests: TestCase {
+class AuthIntegrationTests: LiveDatabaseTestCase {
```
Comment (mbrandonw) [inline Tests/PointFreeTests/AuthTests.swift]: small potential gotcha for having two test classes in a single file is that snapshots may overwrite each other if the test function has the same name. only the #file and #function is baked into the snapshot file name.

File: Tests/PointFreeTests/AccountTests/UpdateProfileTests.swift (hunk 1 above)
Comment (stephencelis) [reply Tests/PointFreeTests/AccountTests/UpdateProfileTests.swift]: Ha yeah.

File: Tests/PointFreeTests/AuthTests.swift (hunk 2 above)
Comment (stephencelis) [reply Tests/PointFreeTests/AuthTests.swift]: Yeah...though I think it's OK since collisions will get caught in test runs?

**Comment by stephencelis** [comment]
Other than the update profile mismatch I believe everything's audited!

**Pull request merged**